src/main/java/rotateit/domain/Event.java

```diff
@@ -0,0 +1,16 @@
+package rotateit.domain;
+
+import org.springframework.context.ApplicationEvent;
+
+@SuppressWarnings("unchecked")
+public class Event<T> extends ApplicationEvent {
```

Comment:
I would create separate event classes for different events. There's not going to be many of them and it will definitely be easier to understand